include/xprec/ddouble-header-only.h

```diff
@@ -0,0 +1,30 @@
+/* Small double-double arithmetic library.
+ *
+ * Copyright (C) 2023 Markus Wallerberger and others
+ * SPDX-License-Identifier: MIT
+ */
+#pragma once
+
+//Visual studio won't define M_PI unless _USE_MATH_DEFINES is defined, which we can't ensure with header only inclusion.
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+    #define CLEAN_M_PI
+#endif
+
+//The strategy for header only use of ddouble is pretty simple:
+//conditionally inline globally defined functions and include the cxx files directly.
+#define INLINE_IF_HEADER_ONLY inline
```

Comment:
These functions are unlikely to be inlined, so it is a bit strange to mark them `inline`. Would it not be better to mark them `static` or would this cause some problems?

Also, I think this macro should be prefixed with `_XPREC_` or something, since it is "private".